### PR TITLE
Update README with development timeline

### PR DESCRIPTION
**Pull request by LLFourn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,74 +2,88 @@
...
 
-1. You can have **both** a feerate and absolute fee constraint.
-2. The coin selection "algorithm" logic does not need to keep track of whether feerate has been satisfied yet etc. All this logic is done for you. (at least in coin selection algorithms that are greedy I don't know how well this idea will work with branch and bound).
-3. No traits needed to be implemented to do coin selection. This is good because you can use bespoke application data like utxo labels etc without having to pass them into something implementing `CoinSelectionAlgorithm`.
-4. `CoinSelector` tries checks if it complete at any stage both with and without change. In bdk the choice of change [is done after](https://github.com/bitcoindevkit/bdk/issues/147) coin selection which is sub-optimal.
+This component is about properly using miniscript's potential to know exactly which outputs you can
+spend and the most efficient way to spend them *before* trying to spend them. This allows you to
+know precisely how much weight spending an output will add for coin selection. This is an important
+feature in a taproot world sine different spending paths can have vastly different weight.
```
Comment (notmandatory) [inline README.md:42]: typo fix:
```suggestion
feature in a taproot world since different spending paths can have vastly different weight.
```

**Comment by notmandatory** [comment]
This looks great and will come in very handy next week for our on-site chats with the Spiral/LDK team. I hope you don't mind if I pull some of this info into slides (with a link back to this README).

**Comment by notmandatory** [comment]
Was there some TODO work left for the file based persistence module ? I think Monday you mentioned something you were still working on there.

**Review by evanlinjin — APPROVED** [review]
Looks good! However, I think it would be good to include more information about our plan with CPFP and transaction building. Will make a separate PR for that!

**Pull request merged**